docs/src/reference/the-traversal.asciidoc

```diff
@@ -2302,6 +2302,281 @@ link:++https://tinkerpop.apache.org/javadocs/x.y.z/core/org/apache/tinkerpop/gre
+source and target vertex IDs, as a parameter.  An ID and a label may also be provided but
+this is optional. If the `T.id` or `T.label` are provided, they need to be enclosed in
+parenthesis. The `mergeE` step can be used directly from the GraphTraversalSource - `g`,
+or in the middle of a traversal. For a match with an existing vertex to occur, all values
+in the map must exist on a vertex; otherwise, a new vertex will be created. The examples
+that follow show how `mergeE` can be used to add relationships between dogs in the graph.       
+
+[gremlin-groovy]
+----
+g.mergeV([(T.id):1,(T.label):'Dog','name':'Toby'])
+g.mergeV([(T.id):2,(T.label):'Dog','name':'Brandy']) <1>
+g.mergeE([(T.label):'Sibling','created':'2022-02-07',(Direction.from):1,(Direction.to):2]) <2>
+g.E().elementMap()
+----
+
+<1> Create two vertices with ID values of 1 and 2.
+<2> Create a "Sibling" relationship between the vertices.
+
+For a `mergeE` step to succeed, both the `from` and `to` vertices must already exist. It
+is not possible to create new vertices directly using `mergeE`, but `mergeV` and `mergeE`
+steps can be combined, in a single query, to achieve that goal.
+
+NOTE: The `mergeE` step will not create vertices that do not exist. In those cases an
+error will be returned. 
+
+If the `Direction` enum has been statically included, its explicit use can be omitted from
+the query.
+
+[gremlin-groovy]
+----
+g.mergeV([(T.id):1,(T.label):'Dog','name':'Toby'])
+g.mergeV([(T.id):2,(T.label):'Dog','name':'Brandy'])
+g.mergeE([(T.label):'Sibling','created':'2022-02-07',(from):1,(to):2])
+g.E().elementMap()
+----
+
+One or more `option` steps can be used to control the behavior when an edge is created of
+updated.
+
+[gremlin-groovy]
+----
+g.mergeV([(T.id):1,(T.label):'Dog','name':'Toby'])
+g.mergeV([(T.id):2,(T.label):'Dog','name':'Brandy'])
+
+g.withSideEffect('map',[(T.label):'Sibling',(from):1,(to):2]).
+  mergeE(select('map')).
+    option(Merge.onCreate,['created':'2022-02-07']).  <1>
+    option(Merge.onMatch,['updated':'2022-02-07'])    <2>
+g.E().elementMap()
+----
+<1> The edge did not exist - set the created date.
+<2> The edge did exist - set the updated date.         
+
+
+More than one edge can be created by a single `mergeE` operation. This is done by
+injecting a list of maps into the traversal and letting them stream into the `mergeE`
+step.
+
+[gremlin-groovy]
+----
+maps=
+  [[(T.label):'Siblings',(from):1,(to):2],
+   [(T.label):'Siblings',(from):1,(to):3]]
+
+g.mergeV([(T.id):1,(T.label):'Dog','name':'Toby'])  <1>
+g.mergeV([(T.id):2,(T.label):'Dog','name':'Brandy'])
+g.mergeV([(T.id):3,(T.label):'Dog','name':'Dax'])
+
+g.inject(maps).unfold().mergeE() <2>
+g.E().elementMap()
+----
+
+<1> Create three dogs.
+<2> Stream the edge maps into `mergeE` steps.
+
+
+[[mergevertex-step]]
+=== MergeVertex Step
+
+The `mergeV()` -step is used to add vertices and their properties to a graph in a "create
```

Comment:
similar feedback here to what I already wrote in `mergeE()` regarding "upgrade docs" and the link to `mergeE()` in the NOTE.